File: 14-TypeScript/Grey Multi-Tenant/packages/grey-r/tests/testthat/test-error_codes.R
test_that("error_code_from_status returns correct codes", {
  expect_equal(error_code_from_status(401), "unauthorized")
  expect_equal(error_code_from_status(403), "forbidden")
  expect_equal(error_code_from_status(404), "not_found")
  expect_equal(error_code_from_status(400), "validation_error")
  expect_equal(error_code_from_status(422), "validation_error")
  expect_equal(error_code_from_status(500), "server_error")
  expect_equal(error_code_from_status(418), "unknown")
})

test_that("is_valid_error_code works", {
  expect_true(is_valid_error_code("unauthorized"))
  expect_true(is_valid_error_code("forbidden"))
  expect_false(is_valid_error_code("invalid_code"))
})

test_that("normalize_error_code normalizes correctly", {
  expect_equal(normalize_error_code("forbidden"), "forbidden")
  expect_equal(normalize_error_code("invalid"), "unknown")
})

File: 14-TypeScript/Grey Multi-Tenant/packages/grey-julia/test/test_error_codes.jl
using Test
using GreySDK.ErrorCodes

@testset "ErrorCodes" begin
    @testset "from_http_status" begin
        @test from_http_status(401) == UNAUTHORIZED
        @test from_http_status(403) == FORBIDDEN
        @test from_http_status(404) == NOT_FOUND
        @test from_http_status(400) == VALIDATION_ERROR
        @test from_http_status(422) == VALIDATION_ERROR
        @test from_http_status(408) == TIMEOUT
        @test from_http_status(504) == TIMEOUT
        @test from_http_status(500) == SERVER_ERROR
        @test from_http_status(502) == SERVER_ERROR
        @test from_http_status(503) == SERVER_ERROR
        @test from_http_status(200) == UNKNOWN
        @test from_http_status(301) == UNKNOWN
    end
    
    @testset "is_valid" begin
        @test is_valid(UNAUTHORIZED) == true
        @test is_valid(FORBIDDEN) == true
        @test is_valid(NOT_FOUND) == true
        @test is_valid(VALIDATION_ERROR) == true
        @test is_valid(NETWORK_ERROR) == true
        @test is_valid(TIMEOUT) == true
        @test is_valid(SERVER_ERROR) == true
        @test is_valid(UNKNOWN) == true
        @test is_valid(:invalid_code) == false
    end
    
    @testset "normalize" begin
        @test normalize(UNAUTHORIZED) == UNAUTHORIZED
        @test normalize(:invalid_code) == UNKNOWN
    end
end
